packages/dashboard/src/components/task-summary-panel.tsx

```diff
@@ -0,0 +1,221 @@
+    setExpanded(nodeIds);
+  };
+
+  const handleSelect = (event: React.ChangeEvent<{}>, nodeIds: string) => {
+    setSelected(nodeIds);
+  };
+
+  const handleResetTasks = () => {
+    setExpanded([]);
+    setSelected('');
+  };
+
+  const handleClearAllCurrTasks = () => {
+    savedTasksContent.current = taskContents;
+    setTaskContents({});
+  };
+
+  const handleRestoreTasks = () => {
+    // Assigning another reference
+    let taskContentsTemp = Object.assign({}, taskContents);
+    Object.keys(savedTasksContent.current).forEach((element) => {
+      if (!(element in taskContents)) {
+        taskContentsTemp[element] = savedTasksContent.current[element];
+      }
+    });
+    setTaskContents(taskContentsTemp);
+    savedTasksContent.current = {};
+  };
+
+  const determineStyle = (state: number): string => {
+    switch (state) {
+      case RomiCore.TaskSummary.STATE_QUEUED:
+        return classes.queued;
+      case RomiCore.TaskSummary.STATE_ACTIVE:
+        return classes.active;
+      case RomiCore.TaskSummary.STATE_COMPLETED:
+        return classes.completed;
+      case RomiCore.TaskSummary.STATE_FAILED:
+        return classes.failed;
+      default:
+        return 'UNKNOWN';
+    }
+  };
+
+  // Update Task list content
+  React.useEffect(() => {
+    const renderActor = (taskStatus: string): React.ReactElement | undefined => {
+      const actor = TaskManager.getActorFromStatus(taskStatus);
+      if (!actor) return;
+      return (
+        <Typography variant="body1" className={classes.taskActor}>
+          {actor}
+        </Typography>
+      );
+    };
+
+    const renderTaskTreeItem = (task: RomiCore.TaskSummary) => {
+      return (
+        <TreeItem
+          data-component="TreeItem"
+          nodeId={task.task_id}
+          key={task.task_id}
+          classes={{
+            label: `${determineStyle(task.state)} ${classes.labelContent}`,
+            root: classes.treeChildren,
+            expanded: classes.expanded,
+          }}
+          label={
+            <>
+              <Typography variant="body1" noWrap>
+                <IconButton
+                  // TODO: replace onClick with the pause plans logic.
+                  onClick={(e) => {
+                    e.preventDefault();
+                    console.log('Not implemented');
+                  }}
+                >
+                  <PlayCircleFilledWhiteIcon />
+                </IconButton>
+                {task.task_id}
+              </Typography>
+              {renderActor(task.status)}
+            </>
+          }
+        >
+          <TaskSummaryPanelItem task={task} key={task.task_id} />
+        </TreeItem>
+      );
+    };
+
+    if (allTasks.length === 0) return;
+    let taskContentsTemp = Object.assign({}, taskContents);
+    allTasks.forEach((task) => {
+      taskContentsTemp[task.task_id] = renderTaskTreeItem(task);
+    });
+    setTaskContents(taskContentsTemp);
```

Comment:
You should be able to use the functional form to avoid recursive dependencies and use of a temp object.
```tsx
setTaskContents((currentContents) => {
  return doStuff(currentContents);
});
```

Reply:
I looked into it and we cannot assign directly the stored values to `currentContent` because it is updating the taskContents too, so when we set the taskContent new value with `setTaskContents` it'll no re-render because it'll not detect any changes.
We can also force the re-render of the component